Apple/App/Menu/MenuView.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  MenuView.swift
+//  App
+//
+//  Created by Thomas Stubblefield on 5/13/23.
+//
+
+import SwiftUI
+
+struct MenuView: View {
+    @State private var isToggled = false
+    @ObservedObject var tunnel: Tunnel
+
+    private func start() {
+        
+        do {
+            try tunnel.start()
+        } catch {
+            print(error)
+        }
+    }
+
+    private func stop() {
+        tunnel.stop()
+    }
+
+    private func configure() {
+        Task { try await tunnel.configure() }
+    }
+
+    var body: some View {
+        VStack {
+            HStack {
+                Text("Burrow")
+                    .fontWeight(.bold)
+
+                Spacer()
+                Toggle("", isOn: $isToggled)
+                    .toggleStyle(SwitchToggleStyle(tint: .blue))
+                    .onChange(of: isToggled) { value in
+                        if value {
```

Comment:
One pattern I try to conform to when writing SwiftUI is "Model-View-ViewModel". In this case, the view is `MenuView`, the view model is `Tunnel` and the model is the underlying system APIs it configures.

In the MVVM architecture, the goal is to keep these components separate from one another for better readability. In the case of MVVM any data manipulation should go inside of the view model itself (Tunnel).

I think what we want to do here is to bring `isToggled` into `Tunnel`:

```swift
extension Tunnel {
    var isOn: Binding<Bool> {
        Binding {
            switch self.status {
            case .connected, .connecting:
                return true
            default:
                return false
            }
        } set: { newValue, _ in
            if newValue {
                try? self.start()
            } else {
                try? self.stop()
            }
        }
    }
}
```

But there are more than two states (on and off), so we might need other ways to indicate that (perhaps disabling the switch when connecting, or showing a spinning indicator instead)

Reply:
> Model-View-ViewModel
JFC 😩